Lesson19/elifcetin/script.js

```diff
@@ -0,0 +1,104 @@
+/*
+===========================================================
+  SHOPPING CART APPLICATION
+===========================================================
+In this project, you'll create a simple Shopping Cart to
+simulate adding items, removing items, calculating totals,
+and applying discounts.
+
+You'll practice:
+1. Classes and objects
+2. Encapsulation and abstraction
+3. Methods (functions inside a class)
+4. Arrays and basic array methods (push, filter, find)
+5. Conditional statements (if-else)
+
+Below is a step-by-step guide with comments explaining
+each part. You can test each step by running the code in
+Node.js or a browser console.
+*/
+
+
+class ShoppingCart {
+  // a class is template that keeps two things together. Data = the information an object has. Method = the actions an object can do.
+  #items; // a property named #items was declared within the class. # makes private class field. it can only be accessed or modified from inside the ShoppingCart class itself, not from outside code.
+  constructor() {
+    //the constructor runs automatically every time you create a new object from the class. Its job is to set up the object—like preparing everything it needs before it can be used.
+    /* Every time you create a new ShoppingCart, it starts with
+    #items = []
+    So each cart begins fresh, empty array ([]). with no items inside.
+   */
+    this.#items = []; // This sets the #items property of the newly created object to an empty array ([]), ensuring every new cart starts with no items
+  }
+  viewCart() {
+    // a method was defined to perform an action in order to display contents. IT HAS ACCESS to private #items array and prints each item to the console
+    for (const item of this.#items) {
+      //iterating over the elements of an object. you get the item's value. Because viewCart only needs to display the items, not know their position. "For every item of the array #items, do something."
+      console.log(item);
+    }
+  }
```

Comment:
Excessive comments can make the code harder to scan. Prefer self-explanatory variable and method names, and use comments only where the intent is not obvious.

```suggestion
class ShoppingCart {
  #items;
  constructor() {
    this.#items = []; 
    }
  viewCart() {
    for (const item of this.#items) {
      console.log(item);
    }
  }
```